### DISCUSSION TITLE
How much history does ChatGPT respect?

### DISCUSSION THREAD

**Discussion by ElRakiti** [spdustin/ChatGPT-AutoExpert#74] [Ideas]
### Edition
Check one or more:
- [ x] Standard Edition
- [ ] Developer Edition

This repo / the way how you use custom instructions are amazing! I currently plan a P&P campaign for my two kids and wife with AutoExpert, building the characters, plotting a story etc. As you can imagine, this generates quite a lot of text, not only the story etc. itself, but also when I deep dive into something specific (e.g. an NPC which occurs right in the beginning of the story) and ChatGPT then describes the character and so on. I'm wondering now, how much of the history (e.g. the initial character stats or the base facts of the story which were decribed in the very beginning) ChatGPT remembers? Is there any limit and what would be the best way to preserve everything so that (e.g.) I can continue with the next campaign which directly continues the first campaign?

Again, awesome work!

**Comment by davidkurland** [comment]
in a similar vein, im curious whether you can apply the architecture of memgpt to this project, to allow autodev to manage its own memory. 
or even baking in something like this
https://github.com/daveshap/SparsePrimingRepresentations

**Comment by rubywwwilde** [comment]
@davidkurland have you tried SPR yourself? was it good for you?

**Comment by spdustin** [comment]
@ElRakiti Developer Edition does its best to manage long-term memory by (basically) summarizing its history at the end of each response, maybe you could pull something together from those custom instructions?

**Comment by spdustin** [comment]
@davidkurland That's an interesting idea... The next version already bundles a wheel for `treesitter` and language configs, so packaging another dependency isn't that big of a deal.

**BUT** (and this is a big but) because ChatGPT's interface with the `python` is via Jupyter Cells that have their own context, it would mean ChatGPT responses taking even longer to finish, since they'd have to write code to store an object in memory on each conversation turn.

**Comment by spdustin** [comment]
Gonna move this to discussions...
